src/models/attendance/attendance-data-mapper-impl.ts

```diff
@@ -112,34 +113,14 @@ export class AttendanceDataMapperImpl extends GenericDataMapper
    * @return {Promise<Stream>}
    */
   public async getAll(opts?: IUowOpts): Promise<IDbResult<Attendance[]>> {
-    let queryBuilder = squel.select({
-      autoQuoteFieldNames: true,
-      autoQuoteTableNames: true,
-    })
-      .from(this.tableName, 'attendance');
+    let queryBuilder = await this.getAttendanceStatQuery(opts);
     let checkCache = true;
     if (opts && opts.ignoreCache) {
       checkCache = false;
     }
     if (opts && opts.fields) {
```

Comment:
Should this have been removed like in the other functions?

Reply:
attendanceByX depends on specific fields which this might override. in case I call getAttendanceByEvents() and pass in the wrong fields, the server will crash. That is why it was removed from the other function. Since this is only passing the data forward, it shouldn't be an issue.